ORT server tries to insert license_findings without score

The column `score` in the table `license_findings` is declared as `NOT NULL`:
https://github.com/eclipse-apoapsis/ort-server/blob/aaed6a9c4d6a04eb504e16b418d10d5af1a320ec/dao/src/main/resources/db/migration/V23__scanResults.sql#L9-L18

But the field `score` in the `LicenseFindingsTable` and the `LicenseFindingDao` are declared as `nullable()`:
https://github.com/eclipse-apoapsis/ort-server/blob/aaed6a9c4d6a04eb504e16b418d10d5af1a320ec/dao/src/main/kotlin/tables/LicenseFindingsTable.kt#L33-L40

This causes the issue that the server tries to insert a license finding, but fails due to the not null constraint:
```
INSERT INTO license_findings (end_line, license, "path", scan_summary_id, start_line) VALUES (?, ?, ?, ?, ?)
INSERT INTO license_findings (end_line, license, "path", scan_summary_id, start_line) VALUES (?, ?, ?, ?, ?);
  Detail: Failing row contains (6958879, Apache-2.0, gradlew, -1, -1, null, 50227).  Call getNextException to see other errors in the batch.. Statement(s): INSERT INTO license_findings (end_line, license, "path", scan_summary_id, start_line) VALUES (?, ?, ?, ?, ?);
RETURNING * was aborted: ERROR: null value in column "score" of relation "license_findings" violates not-null constraint
2024-04-16 14:46:36.326 [main] level=WARN  Exposed - Transaction attempt #2 failed: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO license_findings (end_line, license, "path", scan_summary_id, start_line) VALUES (('-1'::int4), ('Apache-2.0'), ('gradlew'), ('50227'::int8), ('-1'::int4))
	... 96 common frames omitted
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2412)
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2725)
  Detail: Failing row contains (6958879, Apache-2.0, gradlew, -1, -1, null, 50227).
Caused by: org.postgresql.util.PSQLException: ERROR: null value in column "score" of relation "license_findings" violates not-null constraint
	... 82 common frames omitted
	at org.jetbrains.exposed.sql.statements.Statement.executeIn$exposed_core(Statement.kt:97)
	at org.jetbrains.exposed.sql.statements.InsertStatement.executeInternal(InsertStatement.kt:18)
	at org.jetbrains.exposed.sql.statements.InsertStatement.executeInternal(InsertStatement.kt:168)
	at org.jetbrains.exposed.sql.statements.InsertStatement.execInsertFunction(InsertStatement.kt:158)
	at org.jetbrains.exposed.sql.statements.jdbc.JdbcPreparedStatementImpl.executeBatch(JdbcPreparedStatementImpl.kt:97)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeBatch(HikariProxyPreparedStatement.java)
	at com.zaxxer.hikari.pool.ProxyStatement.executeBatch(ProxyStatement.java:127)
	at org.postgresql.jdbc.PgPreparedStatement.executeBatch(PgPreparedStatement.java:1733)
	at org.postgresql.jdbc.PgStatement.executeBatch(PgStatement.java:936)
	at org.postgresql.jdbc.PgStatement.internalExecuteBatch(PgStatement.java:912)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:565)
	at org.postgresql.core.v3.QueryExecutorImpl.sendQuery(QueryExecutorImpl.java:1527)
	at org.postgresql.core.v3.QueryExecutorImpl.flushIfDeadlockRisk(QueryExecutorImpl.java:1502)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2145)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2413)
	at org.postgresql.jdbc.BatchResultHandler.handleError(BatchResultHandler.java:165)
```

--- a/dao/src/main/resources/db/migration/V62__licenseFindingsNullableScore.sql
+++ b/dao/src/main/resources/db/migration/V62__licenseFindingsNullableScore.sql
@@ -1,0 +1,2 @@
+ALTER TABLE license_findings
+    ALTER COLUMN score DROP NOT NULL;

